src/webots/vrml/WbNode.cpp

```diff
@@ -972,13 +977,45 @@ void WbNode::writeParameters(WbVrmlWriter &writer) const {
     parameter->write(writer);
 }
 
+bool WbNode::isUrdfLinkRoot() const {
+  if (findSFString("name") || dynamic_cast<WbBasicJoint *>(parent()))
+    return true;
+  return false;
+}
```

Comment:
Here for example to avoid having to include WbBasicJoint you can make this function virtual and return false.
Then in WbMatter and  WbBasicJoint you make it returns true.

Reply:
I put it in `WbBaseNode`, I think it is more general this way.